jobs/gtfs-rt-parser/gtfs_rt_parser.py

```diff
@@ -46,46 +88,99 @@ class RTFile(BaseModel):
     url: int
     tick: pendulum.DateTime
 
-    def hive_path(self, bucket: str):
+    class Config:
+        json_encoders = {
+            Path: lambda p: str(p),
+        }
+
+    @property
+    def timestamped_filename(self):
+        return str(self.path.name) + self.tick.strftime("__%Y-%m-%dT%H:%M:%SZ.pb")
+
+    @property
+    def schedule_path(self):
+        return os.path.join(
+            get_bucket(),
+            "schedule",
+            str(self.tick.replace(hour=0, minute=0, second=0)),
+            f"{self.itp_id}_{self.url}",
+        )
+
+    def data_hive_path(self, bucket: str):
         return os.path.join(
             bucket,
             self.file_type,
             f"dt={self.tick.to_date_string()}",
             f"itp_id={self.itp_id}",
             f"url_number={self.url}",
             f"hour={self.tick.hour}",
```

Comment:
**question**: This idea was actually mentioned to me in an unrelated conversation by @j-meelah -- is there any chance that we want to still partition on minute / second but just have those values constant, as a way of future proofing in case we ultimately end up processing more often? 

She brought this question up in the context of GTFS schedule. Right now, we might be tempted to partition on day only. But I know you're interested in scraping more often, so should I add an hourly partition level as a method of future-proofing against that eventuality?

Reply:
I think the situations are different.

For schedule, I think that's totally reasonable and most accurate because we actually did download the data at an instant, and we don't plan to aggregate up.

For RT, I don't think so. This is aggregated/transformed data, and we can't partition down to the minute/second without blowing up BigQuery. It's the same problem as having a TIMESTAMP with zeros to indicate a DATE. It's misleading if a data consumer sees timestamps and doesn't have the contextual knowledge that the data is aggregated to the hour.

Reply:
Ok, that distinction makes sense. I think we should maybe share that out as a decision that was made here for awareness.